spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -261,6 +266,32 @@ public long count(Class<?> entityClass) {
 				: null;
 	}
 
+	@Override
+	public <T> Slice<?> queryKeysOrEntitiesSlice(StructuredQuery query, Class<T> entityClass, Pageable pageable) {
+		DatastoreResultsIterable<?> results = queryKeysOrEntities(applyPageable(query, pageable), entityClass);
+		return new SliceImpl<>(results.toList(),
```

Comment:
Can we extend `SliceImpl` such that `hasNextPage()` is lazy?
Feel free to leave for another PR though.

Reply:
I don't think it is necessary because if you don't need  `hasNextPage()`, you could just use the `queryKeysOrEntities` method instead.

Reply:
What about the use in repositories?